samcli/local/services/localhost_runner.py

```diff
@@ -0,0 +1,124 @@
+"""Base class for all Services that interact with Local Lambda"""
+
+import logging
+import os
+
+from flask import Response
+
+LOG = logging.getLogger(__name__)
+
+
+class LocalhostRunner(object):
+
+    def __init__(self, lambda_runner, port, host, stderr=None):
+        """
+        Creates a BaseService class
+
+        Parameters
+        ----------
+        lambda_runner samcli.commands.local.lib.local_lambda.LocalLambdaRunner
+            The Lambda runner class capable of invoking the function
+        port int
+            Optional. port for the service to start listening on Defaults to 3000
+        host str
+            Optional. host to start the service on Defaults to '127.0.0.1
+        stderr io.BaseIO
```

Comment:
`stderr` is not used in this class

Reply:
I originally placed this here because it was used in both Local APIGW and Local Lambda. Since it is unused here, I pushed this up the hierarchy.